src_c/cython/pygame/_sdl2/video.pyx

```diff
@@ -1012,12 +1012,16 @@ cdef class Renderer:
             if SDL_RenderSetViewport(self._renderer, NULL) < 0:
                 raise error()
             return
-        cdef SDL_Rect rect
-        if pgRect_FromObject(area, &rect) == NULL:
-            raise TypeError("the argument is not a rectangle or None")
-        if SDL_RenderSetViewport(self._renderer, &rect) < 0:
+
+        cdef SDL_Rect tmprect
+        cdef SDL_Rect *rectptr = pgRect_FromObject(area, &tmprect)
```

Comment:
There are a few other places in this file where the returned pointer from `pgRect_FromObject()` is not being used.

- Image methods
  - `__init__()`
  - `draw()`

- Renderer method
  - `to_surface()`

Reply:
I've updated these ones too. But didn't add tests for these.